truffle/src/main/java/org/jruby/truffle/core/array/ArrayNodes.java

```diff
@@ -920,13 +918,14 @@ protected Object fillFallback(VirtualFrame frame, DynamicObject array, Object[]
     public abstract static class HashNode extends ArrayCoreMethodNode {
 
         @Child private ToIntNode toIntNode;
+        @Child private CallDispatchHeadNode toHashNode;
```

Comment:
@eregon I've updated this to be `@Cached` now and removed the unused SnippetNode